File: csharp/Svix/Stream.cs
// this file is @generated
#nullable enable
using Microsoft.Extensions.Logging;
using Svix.Models;

namespace Svix
{
    public class Stream(SvixClient client)
    {
        readonly SvixClient _client = client;

        public StreamEventType EventType
        {
            get => new StreamEventType(_client);
        }

        public StreamEvents Events
        {
            get => new StreamEvents(_client);
        }

        public StreamSink Sink
        {
            get => new StreamSink(_client);
        }

        public StreamStream Stream
        {
            get => new StreamStream(_client);
        }

        /// <summary>
        /// Get the HTTP sink headers. Only valid for `http` or `otelTracing` sinks.
        /// </summary>
        public async Task<EndpointHeadersOut> SinkHeadersGetAsync(
            string streamId,
            string sinkId,
            CancellationToken cancellationToken = default
        )
        {
            try
            {
                var response = await _client.SvixHttpClient.SendRequestAsync<EndpointHeadersOut>(
                    method: HttpMethod.Get,
                    path: "/api/v1/stream/{stream_id}/sink/{sink_id}/headers",
                    pathParams: new Dictionary<string, string>
                    {
                        { "stream_id", streamId },
                        { "sink_id", sinkId },
                    },
                    cancellationToken: cancellationToken
                );
                return response.Data;
            }
            catch (ApiException e)
            {
                _client.Logger?.LogError(e, $"{nameof(SinkHeadersGetAsync)} failed");

                throw;
            }
        }

        /// <summary>
        /// Get the HTTP sink headers. Only valid for `http` or `otelTracing` sinks.
        /// </summary>
        public EndpointHeadersOut SinkHeadersGet(string streamId, string sinkId)
        {
            try
            {
                var response = _client.SvixHttpClient.SendRequest<EndpointHeadersOut>(
                    method: HttpMethod.Get,
                    path: "/api/v1/stream/{stream_id}/sink/{sink_id}/headers",
                    pathParams: new Dictionary<string, string>
                    {
                        { "stream_id", streamId },
                        { "sink_id", sinkId },
                    }
                );
                return response.Data;
            }
            catch (ApiException e)
            {
                _client.Logger?.LogError(e, $"{nameof(SinkHeadersGet)} failed");

                throw;
            }
        }

        /// <summary>
        /// Updates the Sink's headers. Only valid for `http` or `otelTracing` sinks.
        /// </summary>
        public async Task<EndpointHeadersOut> SinkHeadersPatchAsync(
            string streamId,
            string sinkId,
            HttpSinkHeadersPatchIn httpSinkHeadersPatchIn,
            CancellationToken cancellationToken = default
        )
        {
            httpSinkHeadersPatchIn =
                httpSinkHeadersPatchIn
                ?? throw new ArgumentNullException(nameof(httpSinkHeadersPatchIn));
            try
            {
                var response = await _client.SvixHttpClient.SendRequestAsync<EndpointHeadersOut>(
                    method: HttpMethod.Patch,
                    path: "/api/v1/stream/{stream_id}/sink/{sink_id}/headers",
                    pathParams: new Dictionary<string, string>
                    {
                        { "stream_id", streamId },
                        { "sink_id", sinkId },
                    },
                    content: httpSinkHeadersPatchIn,
                    cancellationToken: cancellationToken
                );
                return response.Data;
            }
            catch (ApiException e)
            {
                _client.Logger?.LogError(e, $"{nameof(SinkHeadersPatchAsync)} failed");

                throw;
            }
        }

        /// <summary>
        /// Updates the Sink's headers. Only valid for `http` or `otelTracing` sinks.
        /// </summary>
        public EndpointHeadersOut SinkHeadersPatch(
            string streamId,
            string sinkId,
            HttpSinkHeadersPatchIn httpSinkHeadersPatchIn
        )
        {
            httpSinkHeadersPatchIn =
                httpSinkHeadersPatchIn
                ?? throw new ArgumentNullException(nameof(httpSinkHeadersPatchIn));
            try
            {
                var response = _client.SvixHttpClient.SendRequest<EndpointHeadersOut>(
                    method: HttpMethod.Patch,
                    path: "/api/v1/stream/{stream_id}/sink/{sink_id}/headers",
                    pathParams: new Dictionary<string, string>
                    {
                        { "stream_id", streamId },
                        { "sink_id", sinkId },
                    },
                    content: httpSinkHeadersPatchIn
                );
                return response.Data;
            }
            catch (ApiException e)
            {
                _client.Logger?.LogError(e, $"{nameof(SinkHeadersPatch)} failed");

                throw;
            }
        }

        /// <summary>
        /// Get the transformation code associated with this sink.
        /// </summary>
        public async Task<SinkTransformationOut> SinkTransformationGetAsync(
            string streamId,
            string sinkId,
            CancellationToken cancellationToken = default
        )
        {
            try
            {
                var response = await _client.SvixHttpClient.SendRequestAsync<SinkTransformationOut>(
                    method: HttpMethod.Get,
                    path: "/api/v1/stream/{stream_id}/sink/{sink_id}/transformation",
                    pathParams: new Dictionary<string, string>
                    {
                        { "stream_id", streamId },
                        { "sink_id", sinkId },
                    },
                    cancellationToken: cancellationToken
                );
                return response.Data;
            }
            catch (ApiException e)
            {
                _client.Logger?.LogError(e, $"{nameof(SinkTransformationGetAsync)} failed");

                throw;
            }
        }

        /// <summary>
        /// Get the transformation code associated with this sink.
        /// </summary>
        public SinkTransformationOut SinkTransformationGet(string streamId, string sinkId)
        {
            try
            {
                var response = _client.SvixHttpClient.SendRequest<SinkTransformationOut>(
                    method: HttpMethod.Get,
                    path: "/api/v1/stream/{stream_id}/sink/{sink_id}/transformation",
                    pathParams: new Dictionary<string, string>
                    {
                        { "stream_id", streamId },
                        { "sink_id", sinkId },
                    }
                );
                return response.Data;
            }
            catch (ApiException e)
            {
                _client.Logger?.LogError(e, $"{nameof(SinkTransformationGet)} failed");

                throw;
            }
        }
    }
}

File: python/svix/api/stream.py
# This file is @generated
from ..models import EndpointHeadersOut, HttpSinkHeadersPatchIn, SinkTransformationOut
from .common import ApiBase
from .stream_event_type import (
    StreamEventType,
    StreamEventTypeAsync,
)
from .stream_events import (
    StreamEvents,
    StreamEventsAsync,
)
from .stream_sink import (
    StreamSink,
    StreamSinkAsync,
)
from .stream_stream import (
    StreamStream,
    StreamStreamAsync,
)


class StreamAsync(ApiBase):
    @property
    def event_type(self) -> StreamEventTypeAsync:
        return StreamEventTypeAsync(self._client)

    @property
    def events(self) -> StreamEventsAsync:
        return StreamEventsAsync(self._client)

    @property
    def sink(self) -> StreamSinkAsync:
        return StreamSinkAsync(self._client)

    @property
    def stream(self) -> StreamStreamAsync:
        return StreamStreamAsync(self._client)

    async def sink_headers_get(
        self, stream_id: str, sink_id: str
    ) -> EndpointHeadersOut:
        """Get the HTTP sink headers. Only valid for `http` or `otelTracing` sinks."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/stream/{stream_id}/sink/{sink_id}/headers",
            path_params={
                "stream_id": stream_id,
                "sink_id": sink_id,
            },
        )
        return EndpointHeadersOut.model_validate(response.json())

    async def sink_headers_patch(
        self,
        stream_id: str,
        sink_id: str,
        http_sink_headers_patch_in: HttpSinkHeadersPatchIn,
    ) -> EndpointHeadersOut:
        """Updates the Sink's headers. Only valid for `http` or `otelTracing` sinks."""
        response = await self._request_asyncio(
            method="patch",
            path="/api/v1/stream/{stream_id}/sink/{sink_id}/headers",
            path_params={
                "stream_id": stream_id,
                "sink_id": sink_id,
            },
            json_body=http_sink_headers_patch_in.model_dump_json(
                exclude_unset=True, by_alias=True
            ),
        )
        return EndpointHeadersOut.model_validate(response.json())

    async def sink_transformation_get(
        self, stream_id: str, sink_id: str
    ) -> SinkTransformationOut:
        """Get the transformation code associated with this sink."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/stream/{stream_id}/sink/{sink_id}/transformation",
            path_params={
                "stream_id": stream_id,
                "sink_id": sink_id,
            },
        )
        return SinkTransformationOut.model_validate(response.json())


class Stream(ApiBase):
    @property
    def event_type(self) -> StreamEventType:
        return StreamEventType(self._client)

    @property
    def events(self) -> StreamEvents:
        return StreamEvents(self._client)

    @property
    def sink(self) -> StreamSink:
        return StreamSink(self._client)

    @property
    def stream(self) -> StreamStream:
        return StreamStream(self._client)

    def sink_headers_get(self, stream_id: str, sink_id: str) -> EndpointHeadersOut:
        """Get the HTTP sink headers. Only valid for `http` or `otelTracing` sinks."""
        response = self._request_sync(
            method="get",
            path="/api/v1/stream/{stream_id}/sink/{sink_id}/headers",
            path_params={
                "stream_id": stream_id,
                "sink_id": sink_id,
            },
        )
        return EndpointHeadersOut.model_validate(response.json())

    def sink_headers_patch(
        self,
        stream_id: str,
        sink_id: str,
        http_sink_headers_patch_in: HttpSinkHeadersPatchIn,
    ) -> EndpointHeadersOut:
        """Updates the Sink's headers. Only valid for `http` or `otelTracing` sinks."""
        response = self._request_sync(
            method="patch",
            path="/api/v1/stream/{stream_id}/sink/{sink_id}/headers",
            path_params={
                "stream_id": stream_id,
                "sink_id": sink_id,
            },
            json_body=http_sink_headers_patch_in.model_dump_json(
                exclude_unset=True, by_alias=True
            ),
        )
        return EndpointHeadersOut.model_validate(response.json())

    def sink_transformation_get(
        self, stream_id: str, sink_id: str
    ) -> SinkTransformationOut:
        """Get the transformation code associated with this sink."""
        response = self._request_sync(
            method="get",
            path="/api/v1/stream/{stream_id}/sink/{sink_id}/transformation",
            path_params={
                "stream_id": stream_id,
                "sink_id": sink_id,
            },
        )
        return SinkTransformationOut.model_validate(response.json())
